python/tk_houdini/base_hooks/node_handler_base.py

```diff
@@ -0,0 +1,471 @@
+        sgtk_folder = hou.FolderParmTemplate(
+            self.SGTK_FOLDER,
+            "SGTK (ver: {})".format(engine_version),
+            parm_templates=sgtk_templates,
+            folder_type=hou.folderType.Simple
+        )
+        return sgtk_folder
+
+    def _set_up_node(self, node, parameter_group, hou=None):
+        """
+        Set up a node for use with shotgun pipeline.
+
+        :param node: A :class:`hou.Node` instance.
+        :param parameter_group: A :class:`ParmGroup` instance.
+        :param hou: The houdini module. We have to lazy load the houdini python
+            module here, but not in the hooks, so use hook's imports and pass it
+            for efficiency.
+        """
+        if not hou:
+            import hou
+        self._set_up_parms(node)
+        sgtk_folder = self._create_sgtk_folder(node)
+        if not self.SGTK_IDENTIFIER in parameter_group:
+            sgtk_identifier = hou.ToggleParmTemplate(
+                self.SGTK_IDENTIFIER,
+                "SGTK",
+                default_value=True,
+                is_hidden=True
+            )
+            parameter_group.append_template(sgtk_identifier)
+        self._customise_parameter_group(node, parameter_group, sgtk_folder)
+        node.setParmTemplateGroup(parameter_group.build())
+
+    def add_sgtk_parms(self, node):
+        """
+        Add sgtk parameters to the given node.
+
+        :param node: A :class:`hou.Node` instance.
+        """
+        parameter_group = self._get_parameter_group(node)
+        if not parameter_group:
+            return
+        self._set_up_node(node, parameter_group)
+
+    #############################################################################################
+    # UI Callbacks
+    #############################################################################################
+    
+    def _enable_sgtk(self, node, sgtk_enabled):
+        """
+        Enable/disable the sgtk parameters.
+
+        :param node: A :class:`hou.Node` instance.
+        :param bool sgtk_enabled: The state to set the parameters to.
+        """
+        pass
+
+    def enable_sgtk(self, kwargs):
+        """
+        Callback to enable/disable the sgtk parameters.
+        """
+        node = kwargs["node"]
+        use_sgtk = node.parm(self.USE_SGTK)
+        value = use_sgtk.eval()
+        self._enable_sgtk(node, value)
+        if value:
+            self.refresh_file_path(kwargs)
+
+    def _get_all_versions(self, node):
+        """
+        Retrieve all the versions stored on the node.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :rtype: list(int)
+        """
+        sgtk_all_versions = node.parm(self.SGTK_ALL_VERSIONS)
+        all_versions_str = sgtk_all_versions.evalAsString()
+        all_versions = filter(None, all_versions_str.split(","))
+        all_versions = map(int, all_versions)
+        return all_versions
```

Comment:
Safer for when Python 3 comes around if you absolutely have to return a list

```suggestion
        return list(all_versions)
```